test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -366,19 +380,109 @@ class test_uct_cm_sockaddr : public uct_test {
+
+        if (!self->reject_conn_request) {
+            self->m_server->accept(conn_request, server_connect_cb,
+                                   server_disconnect_cb, self);
+        } else {
+            status = uct_listener_reject(listener, conn_request);
+            ASSERT_UCS_OK(status);
+            self->m_cm_state |= TEST_CM_STATE_SERVER_REJECTED;
+        }
+    }
+
+    static void
+    server_connect_cb(uct_ep_h ep, void *arg, ucs_status_t status) {
+         test_uct_cm_sockaddr *self;
+
+         self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+         self->m_cm_state |= TEST_CM_STATE_SERVER_CONNECTED;
+         self->server_connect_cb_cnt++;
+    }
+
+    static void
+    server_disconnect_cb(uct_ep_h ep, void *arg) {
+        test_uct_cm_sockaddr *self;
+
+        self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+        self->m_server->disconnect(ep);
+        self->m_cm_state |= TEST_CM_STATE_SERVER_DISCONNECTED;
+        self->server_disconnect_cnt++;
+    }
+
+    static void
+    client_connect_cb(uct_ep_h ep, void *arg,
+                      const uct_cm_remote_data_t *remote_data,
+                      ucs_status_t status) {
+        test_uct_cm_sockaddr *self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+
+        if (status == UCS_ERR_REJECTED) {
+            self->m_cm_state |= TEST_CM_STATE_CLIENT_GOT_REJECT;
+        } else {
+            ASSERT_UCS_OK(status);
+            EXPECT_EQ(entity::server_priv_data.length() + 1, remote_data->conn_priv_data_length);
+            EXPECT_EQ(entity::server_priv_data,
+                      std::string(static_cast<const char *>(remote_data->conn_priv_data)));
+            self->client_connect_cb_cnt++;
+            self->m_cm_state |= TEST_CM_STATE_CLIENT_CONNECTED;
+        }
+
+    }
+
+    static void
+    client_disconnect_cb(uct_ep_h ep, void *arg) {
+        test_uct_cm_sockaddr *self;
+
+        self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+        self->m_cm_state |= TEST_CM_STATE_CLIENT_DISCONNECTED;
+        self->client_disconnect_cnt++;
+    }
+
+    void cm_disconnect(entity *client) {
+        size_t i;
+
+        /* Disconnect all the existing endpoints */
+        for (i = 0; i < client->num_eps(); ++i) {
+            client->disconnect(client->ep(i));
+        }
+
+        wait_for_bits(&m_cm_state, TEST_CM_STATE_CLIENT_DISCONNECTED |
+                                   TEST_CM_STATE_SERVER_DISCONNECTED);
+        EXPECT_TRUE(ucs_test_all_flags(m_cm_state, (TEST_CM_STATE_SERVER_DISCONNECTED |
+                                                    TEST_CM_STATE_CLIENT_DISCONNECTED)));
     }
 
 protected:
     ucs::sock_addr_storage m_listen_addr, m_connect_addr;
-    entity                 *m_server, *m_client;
+    uint64_t               m_cm_state;
+    entity                 *m_server;
+    entity                 *m_client;
+    volatile int           server_recv_req_cnt, client_connect_cb_cnt,
```

Comment:
members start from "m_"
do we really need all these counter? seems like a duplication of `m_cm_state`

Reply:
these counters are used in additional stress tests i wrote but didn't make it to this PR. i'll remove them for now and restore in the next PR along with the new tests.